### ISSUE TITLE
Unable to use docker ps --filter when a label value has a comma

### ISSUE THREAD

**Issue by lmakarov** [docker/docker#30575]
**Description**

An attempt to filter containers by label when the label value has a comma fails with

```
bad format of filter (expected name=value)
```

**Steps to reproduce the issue:**
1. Create a container

    ```
    $ docker create --name test-label --label 'mylabel=value1,value2' busybox
    ```
2. Confirm the label was properly set

    ```
    $ docker inspect test-label | grep -a3 Labels
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "mylabel": "value1,value2"
            }
        },
    ```

3. Try running docker ps and filter on the label

    ```   
    $ docker ps -a --filter "label=mylabel=value1,value2"
    bad format of filter (expected name=value)
    ```

**Describe the results you received:**

Getting an error from `docker ps`

```
bad format of filter (expected name=value)
```

**Describe the results you expected:**

See the respective container in the output list.

**Additional information you deem important (e.g. issue happens only occasionally):**

None

**Output of `docker version`:**

```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Wed Jan 11 00:23:16 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Wed Jan 11 00:23:16 2017
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 22
 Running: 8
 Paused: 0
 Stopped: 14
Images: 15
Server Version: 1.12.6
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 237
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.41-boot2docker
Operating System: Boot2Docker 1.12.6 (TCL 7.2); HEAD : 5ab2289 - Wed Jan 11 03:20:40 UTC 2017
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 995.8 MiB
Name: docksal
ID: Y5RI:OJMO:CITK:FQFZ:OAFZ:5T2D:UGNS:Y6XI:6H4H:2EFX:ZJQN:XNWV
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 75
 Goroutines: 73
 System Time: 2017-01-30T22:45:39.580227457Z
 EventsListeners: 2
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Tested with boot2docker in VirtualBox as well as with Ubuntu 16.04.

**Comment by vdemeester** [comment]
It's fixed in 1.13.0 and master for sure (can't reproduce)

**Comment by lmakarov** [comment]
Yep, I can also confirm this is no longer an issue in 1.13.1.
Going to close as I don't expect this to get fixed for the earlier versions.
